downloader/src/main/kotlin/Downloader.kt

```diff
@@ -384,6 +385,10 @@ class Downloader {
             DownloadResult(startTime, outputDirectory, sourceArtifact = pomArtifact)
         } catch (e: FileNotFoundException) {
             throw DownloadException("Failed to download the Maven POM for '${target.id}'.")
+        } catch (e: MalformedURLException) {
+            // TODO: Investigate why the binary artifact URL is actually empty and update
+            // the implementation according to root cause.
```

Comment:
A `MalformedURLException` does not mean that the URL has to be empty, it could also be malformed in a different way.
IMO it would be cleaner to use our extension function `String.isValidUrl()` somewhere before line 376 to create a better error message. If we know the URL could be malformed, why run into an exception?
Regarding "update the implementation according to the root cause" I think the downloader should only care about if the URL is valid, not why, so even if the root cause is fixed in the analyzer (if possible) this exception should still be handled here.